cmake/setup_cached_variables.cmake

```diff
@@ -129,6 +129,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
+# Unfortunately CMake doesn't support custom configuration types so we just
+# call "DebugRelease" a "Release" build as well.
+#
+IF(CMAKE_BUILD_TYPE MATCHES "Release")
+  SET(CMAKE_CONFIGURATION_TYPES "Release" CACHE STRING "" FORCE)
+ELSE()
+  SET(CMAKE_CONFIGURATION_TYPES "Debug" CACHE STRING "" FORCE)
```

Comment:
Does the choice of the type have any consequence? I am asking because I have to specify a build type when running tests on windows in ctest. I have to try what happens with this change (or if it would be easier if it is always "debug", or if I need something similar for each test target, etc.)